cranelift/isle/isle/src/sema.rs

```diff
@@ -1166,7 +1290,7 @@ impl TypeEnv {
         self.errors.push(err);
     }
 
-    fn intern_mut(&mut self, ident: &ast::Ident) -> Sym {
+    fn get_or_intern(&mut self, ident: &ast::Ident) -> Sym {
```

Comment:
I'm not sure I like this rename -- semantically I think of interning as the action of getting the existing ID, or creating a new one -- that is, "to intern" covers both cases; and `intern_mut` indicates we're permitting new cases to update the table, vs. `intern` that only looks up existing cases.

(To see this another way, consider `intern` below -- it should be renamed to something else if we take this rename, because this rename implies that adding to the table is the "intern" bit, which is exactly the bit that the non-mut version below does not do.)